### ISSUE TITLE
Game, recruitment of opponent heroes improved by AI players

### ISSUE THREAD

**Issue by LeHerosInconnu** [ihhub/fheroes2#2498]
In fheroes2, it's too easy to recruit heroes improved by AI players compared to the original game.

In this game, I have already recruited heroes from the opposing players and there is still one hero available for recruitment in the tavern.

In fheroes2:

![Easy AI hero fh2 01](https://user-images.githubusercontent.com/43583994/103178186-e29ebe00-4880-11eb-82ce-f99b2e8224cc.png)

![Easy AI hero fh2 02](https://user-images.githubusercontent.com/43583994/103178187-e599ae80-4880-11eb-8759-90acea1600dc.png)

Here is the save file:
[Enemy hero tavern easy fh2 01.zip](https://github.com/ihhub/fheroes2/files/5745058/Enemy.hero.tavern.easy.fh2.01.zip)

**Comment by vincent-grosbois** [comment]
I don't get it, what's the intended behavior?

**Comment by LeHerosInconnu** [comment]
Hello @vincent-grosbois,

> I don't get it, what's the intended behavior?

In the original game, AI players seem to recruit their defeated heroes more often, so the human player is less likely to be able to recruit them.
Someone should check the source code of the original game to see how this is handled.

**Comment by Branikolog** [comment]
I agree with @LeHerosInconnu. Enemy heroes with high level or with artifacts seem to appear in tavern too often. Either AI doesn't buy heroes that had run from a battle, or AI heroes were defeated too often...

**Comment by idshibanov** [comment]
Currently AI buys hero only at the start of the turn, have to add another check at the end after all moves.

**Comment by MicroWombat** [comment]
I keep running into this in version 1.1.16 as well: on almost every map, it’s possible very early on — sometimes literally in week 2 — to hire a leveled-up hero whom the previous owner somehow forgot about after they died.
